File: inc.graphql-resources/backend/python/models/core/message.py
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from database import Base
from datetime import datetime

class Message(Base):
    __tablename__ = 'message'

    message_id = Column(String(40), primary_key=True)
    message_folder_id = Column(String(40), ForeignKey('message_folder.message_folder_id'))
    sender_id = Column(String(40), ForeignKey('admin.admin_id'))
    receiver_id = Column(String(40), ForeignKey('admin.admin_id'))
    subject = Column(String(255), nullable=False)
    content = Column(Text)
    time_create = Column(DateTime, default=datetime.utcnow)
    is_read = Column(Boolean, default=False)
    time_read = Column(DateTime)

    sender = relationship("Admin", foreign_keys=[sender_id])
    receiver = relationship("Admin", foreign_keys=[receiver_id])

File: inc.graphql-resources/backend/nodejs/models/core/message.js
module.exports = (sequelize, DataTypes) => {
    const Message = sequelize.define('message', {
        message_id: {
            type: DataTypes.STRING(40),
            primaryKey: true,
            allowNull: false
        },
        message_direction: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        icon: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        // sender_id, receiver_id, message_folder_id are already defined
        subject: {
            type: DataTypes.STRING(255),
            allowNull: true
        },
        content: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        sender_id: {
            type: DataTypes.STRING(40),
            allowNull: true // Can be from 'system'
        },
        receiver_id: {
            type: DataTypes.STRING(40),
            allowNull: false
        },
        message_folder_id: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        is_read: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        },
        time_create: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        },
        time_read: {
            type: DataTypes.DATE,
            allowNull: true
        },
        ip_read: {
            type: DataTypes.STRING(50),
            allowNull: true
        }
    }, {
        tableName: 'message',
        timestamps: false
    });

    Message.associate = (models) => {
        Message.belongsTo(models.admin, { as: 'sender', foreignKey: 'sender_id' });
        Message.belongsTo(models.admin, { as: 'receiver', foreignKey: 'receiver_id' });
        Message.belongsTo(models.message_folder, { as: 'folder', foreignKey: 'message_folder_id' });
    };

    return Message;
};